birdhouse/components/README.rst

```diff
@@ -135,6 +135,131 @@ Features missing in old install scripts or how the new mechanism improves on the
   via ``env.local``, which should be source controlled, meaning all surrounding maintenance related tasks can also be
   traceable and reproducible.
 
+How to test platform autodeploy is not broken by a PR
+-----------------------------------------------------
+
+There are 2 tests that need to be performed:
+
+* Can autodeploy deploy the PR from ``master`` branch, the stable reference point
+
+  * This could fail if some changes in the PR, are incompatible with autodeploy. For example: ``./pavics-compose.sh`` calls some binaries that do not exist in the autodeploy docker image.
+
+* Can autodeploy be triggered again successfully, after the PR is live
+
+  * This could fail if the PR renamed some files and forgot to add the old file names to a ``.gitignore`` file.  Then old file names will appear as new uncommitted files and autodeploy will halt because it expects a clean working directory.
+
+Here is a sample setup to test autodeploy:
+
+* Have 2 checkout directories.  One is for starting the stack using ``./pavics-compose.sh``, the other one is to push new bogus changes to trigger the autodeploy mechanism.
+
+.. code-block:: shell
+
+  # this one for running pavics-compose.sh
+  git clone git@github.com:bird-house/birdhouse-deploy.git birdhouse-deploy
+
+  # this one for triggering autodeploy
+  git clone git@github.com:bird-house/birdhouse-deploy.git birdhouse-deploy-trigger
+
+* Set ``AUTODEPLOY_PLATFORM_FREQUENCY`` in ``env.local`` to a very frequent value so you do not have to wait too long for autodeploy to trigger.
+
+.. code-block:: shell
+
+  # go to the main checkout
+  cd birdhouse-deploy/birdhouse
+
+  # ensure the scheduler component is enabled, otherwise autodeploy will not work
+  echo 'export EXTRA_CONF_DIRS="$EXTRA_CONF_DIRS ./components/scheduler" >> env.local
+
+  # set AUTODEPLOY_PLATFORM_FREQUENCY
+  echo 'export AUTODEPLOY_PLATFORM_FREQUENCY="@every 5m"' >> env.local
```

Comment:
Hehe, because you have a powerful machine :D  On slower machine, stop and start takes longer, especially when you have many extra components enabled (I have 2 thredds and 3 geoservers !).

I can add a note that the limit can be lower if the machine is powerful enough.